v1/src/main/java/com/google/cloud/teleport/spanner/AvroRecordConverter.java

```diff
@@ -579,6 +628,27 @@ private static Optional<Boolean> readBool(
     }
   }
 
+  private static Optional<Float> readFloat32(
+      GenericRecord record, Schema.Type avroType, String fieldName) {
+    switch (avroType) {
+      case INT:
+        // Can cause loss of precision: un-representable integers will be rounded to the nearest
+        // float.
+        return Optional.ofNullable((Integer) record.get(fieldName)).map(x -> (float) x);
+      case FLOAT:
+        return Optional.ofNullable((Float) record.get(fieldName));
+      case STRING:
+        return Optional.ofNullable((Utf8) record.get(fieldName))
+            .map(Utf8::toString)
+            .map(Float::valueOf);
+        // Avoid decoding 64 bit values into 32 bit space as this will cause a precision loss.
+      case LONG:
+      case DOUBLE:
```

Comment:
What is the template behaviour if the user is trying to perform this conversion?

Reply:
I'm not sure what the template would do -- I think the exception would be caught and the template would fail. 

The alternative option would be to not fail these cases, which will result in silent conversion of INT64/FLOAT64 values to `Infinity`, which would cause unexpected data loss.

I felt that the safer path would be to fail the conversion instead of proceeding with importing lossy data. If we really want to support them, we could possibly have a flag/option that the customer can willingly set to override this failure.

FLOAT32 is the first type that sees this issue, as we were supporting LONG and DOUBLE, which support storing the smaller types (INT and FLOAT). If and when we start supporting INT32, we would have to take the same decision on whether to support conversions from a higher range type to a lower range one.

Reply:
This goes back to the earliest comment around if "rounding and lossy conversions are acceptable" or not.

Reply:
Going with the safer option of loss of precision not being acceptable. We could revisit this without breaking backward compatibility if we want to change our stance

Reply:
Same comment as in the earlier conversation. We do not support transformations in Import/Export. So, I'm okay with throwing an exception on lossy conversion and handling it later if required.

Reply:
Thanks!